src/Swift.Bindings/src/Model/TypeNames/SwiftTypeName.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents a Swift type name.
+/// </summary>
+public record SwiftTypeName
+{
+    /// <summary>
+    /// The module name.
+    /// </summary>
+    public string Module { get; }
+
+    /// <summary>
+    /// The type name.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// The module-qualified type name. Includes all parent types.
+    /// </summary>
+    public string ModuleQualifiedName { get; }
```

Comment:
Is this the unique type name in the type database?

Reply:
Yes - this should represent the full module qualified path to a type